stdpopsim/genomes.py

```diff
@@ -150,16 +204,162 @@ class Contig:
     genetic_map = attr.ib(default=None, kw_only=True)
     inclusion_mask = attr.ib(default=None, kw_only=True)
     exclusion_mask = attr.ib(default=None, kw_only=True)
+    genomic_element_types = attr.ib(type=list, kw_only=True)
+    mutation_types = attr.ib(type=list, kw_only=True)
+
+    @property
+    def length(self):
+        if self.recombination_map is not None:
+            return self.recombination_map.get_length()
+        return None
+
+    @property
+    def is_slim_incompatible(self):
```

Comment:
stdpopsim was not built to simulate continuous stuff anyway, so just one check `has_recombination_map`.